programs/bpf_loader/src/syscalls.rs

```diff
@@ -2118,976 +2285,3475 @@ fn call<'a>(
+    loader_id: &Pubkey,
+    buffers: &mut Vec<BigNum>,
+    bytes: &mut u64,
+    memory_mapping: &MemoryMapping,
+    result: &mut Result<u64, EbpfError<BpfError>>,
+) {
+    let args = question_mark!(
+        translate_slice::<&[u8]>(memory_mapping, arg_addr, arg_count, loader_id, true,),
+        result
+    );
+    *bytes = 0;
+    // BigNums are all but ast
+    for arg in args[0..args.len() - 1].iter() {
+        buffers.push(
+            BigNum::from_slice(question_mark!(
+                translate_slice::<u8>(
+                    memory_mapping,
+                    arg.as_ptr() as u64,
+                    arg.len() as u64,
+                    loader_id,
+                    true
+                ),
+                result
+            ))
+            .unwrap(),
         )
-        .unwrap();
+    }
+    // Negative flags are last
+    let last = args.last().unwrap();
+    let neg_flags = question_mark!(
+        translate_slice::<u8>(
+            memory_mapping,
+            last.as_ptr() as u64,
+            last.len() as u64,
+            loader_id,
+            true
+        ),
+        result
+    );
+    // Get byte count and set negative flag
+    *bytes += neg_flags.len() as u64;
+    for x in 0..neg_flags.len() {
+        *bytes += buffers[x].num_bytes() as u64;
+        buffers[x].set_negative(neg_flags[x] == 1)
+    }
+}
 
-        let cases = vec![
-            (true, START, 0, addr),
-            (true, START, 1, addr),
-            (true, START, LENGTH, addr),
-            (true, START + 1, LENGTH - 1, addr + 1),
-            (false, START + 1, LENGTH, 0),
-            (true, START + LENGTH - 1, 1, addr + LENGTH - 1),
-            (true, START + LENGTH, 0, addr + LENGTH),
-            (false, START + LENGTH, 1, 0),
-            (false, START, LENGTH + 1, 0),
-            (false, 0, 0, 0),
-            (false, 0, 1, 0),
-            (false, START - 1, 0, 0),
-            (false, START - 1, 1, 0),
-            (true, START + LENGTH / 2, LENGTH / 2, addr + LENGTH / 2),
-        ];
-        for (ok, start, length, value) in cases {
-            if ok {
-                assert_eq!(
-                    translate(&memory_mapping, AccessType::Load, start, length).unwrap(),
-                    value
-                )
-            } else {
-                assert!(translate(&memory_mapping, AccessType::Load, start, length).is_err())
+/// Add BigNums
+struct SyscallBigNumAdd<'a> {
+    cost: u64,
+    compute_meter: Rc<RefCell<dyn ComputeMeter>>,
+    loader_id: &'a Pubkey,
+}
+impl<'a> SyscallObject<BpfError> for SyscallBigNumAdd<'a> {
+    fn call(
+        &mut self,
+        arguments_addr: u64,
+        arguments_count: u64,
+        out_result_addr: u64,
+        out_negative_addr: u64,
+        out_size_addr: u64,
+        memory_mapping: &MemoryMapping,
+        result: &mut Result<u64, EbpfError<BpfError>>,
+    ) {
+        let my_func = String::from("sol_bignum_add");
+        match arguments_count == 3 {
```

Comment:
Why not `if`?

Reply:
Because I've still one foot in the `n00b` lake.... changing them to `if`